### PR TITLE
Fix issue 47338 Created a centralized service for links/urls from source code for easy iteration and testing

### PR DESCRIPTION
**Pull request by Emmanuel1999**
Signed-off-by: Emmanuel Opatola <emmanuelopatola99@gmail.com>

Creating a centralized location (service) for source code external links/urls for easy iteration and testing. 

Fixes: https://tracker.ceph.com/issues/47338

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/link-provider.service.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+
+
```
Comment (ljflores) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/services/link-provider.service.ts:3]: Remove this line
```
[lint:tslint  ] /home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/dashboard/frontend/src/app/shared/services/link-provider.service.ts:3:1
[lint:tslint  ] ERROR: 3:1    no-consecutive-blank-lines  Consecutive blank lines are forbidden
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/link-provider.service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return url;
+  };
+
+  telemetrycomponent2 () {
+   const url =  `https://cdla.io/sharing-1.0/`
+  return url;
+  };
+
+  docservice1 () {
+    const url =  "https://docs.ceph.com/en/${docVersion}/"
```
Comment (ljflores) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/services/link-provider.service.ts:22]:
```
[lint:tslint  ] ERROR: 22:18  quotemark                   " should be '
```

**Review by ljflores — COMMENTED** [review]
I know this is a work in progress, but I left a few comments about issues the linter saw with the code. Fixing these will help it pass the Dashboard frontend tests.

**Comment by Emmanuel1999** [comment]
@ljflores I've made the corrections indicated earlier, kindly review thanks

**Pull request closed without merging**